Updated documentation structure for Orange3-Network

ajdapretnar opened the pull request:
- Updated readme
- Changed doc structure to fit the one in Orange3-Text (folder structure, makefile, etc.)
- Added templates for documentation
- Added icons

kernc commented on README.rst:
```diff
@@ -1,12 +1,43 @@
 Documentation is found at:
 
-http://orange-network.readthedocs.org/
+https://readthedocs.org/projects/orange3-network/
```
http://orange3-network.rtfd.io

kernc commented on README.rst:
```diff
@@ -1,12 +1,43 @@
+
+Windows users have to compile the code. Run
+
+	python setup.py build_ext -i --compiler=msvc install
```
Dedent.

ajdapretnar replied:
I've got no idea why this shows such an indent. Looks fine in my editor. :confused:

kernc commented on README.rst:
```diff
@@ -1,12 +1,43 @@
+After the installation, the widgets from this add-on are registered with Orange. To run Orange from the terminal,
+use
+
+    python3 -m Orange.canvas
```
Simply `orange-canvas` should also work.

kernc commented on doc/conf.py:
```diff
@@ -1,95 +1,87 @@
+# ones.
+extensions = [
+    'sphinx.ext.autodoc',
+    'sphinx.ext.intersphinx',
```
What about other extensions that [were specified](https://github.com/biolab/orange3-network/pull/10/files#diff-684a0ceb2878d48b19612ca360e277efL41). Have you ensured they weren't used?

ajdapretnar replied:
Atm there are only templates available, any only one of them is using autodoc extension. Intersphinx isn't even used. So once we add the documentation with meaningful content, we'll update the list of extensions.

kernc replied:
Perhaps add `sphinx.ext.napoleon`. There may be some numpied docstrings. Or they certainly might be in the future.

ajdapretnar replied:
Added `sphinx.ext.napoleon`.

kernc commented on doc/conf.py:
```diff
@@ -1,95 +1,87 @@
+version = '1.1.2'
 # The full version, including alpha/beta/rc tags.
-release = VERSION
+release = '1.1.2'
```
So I kind of like the approach this module was taking, reading the values from _setup.py_. They are defined in a single place which makes it harder for values in one place to go stale.

ajdapretnar replied:
This is build by sphinx. I've got no idea how to handle this otherwise.

ajdapretnar replied:
So if I just change it back to VERSION, it should be fine?

kernc replied:
As you already discovered, _conf.py_ is a simple Python module that features [standard Python code](https://github.com/biolab/orange3-network/pull/10/files#diff-684a0ceb2878d48b19612ca360e277efL23).

kernc commented on doc/conf.py:
```diff
@@ -175,30 +175,59 @@
+#html_search_scorer = 'scorer.js'
+
+# Output file base name for HTML help builder.
+htmlhelp_basename = 'Orange3-Networkdoc'
```
This name is ok?

ajdapretnar replied:
Boh, I guess.

> htmlhelp_basename = 'OrangeVisualProgrammingdoc'

This is all built by Sphinx. It should know............. kind of.

ajdapretnar commented on README.rst:
```diff
@@ -1,12 +1,47 @@
+After the installation, the widgets from this add-on are registered with Orange. To run Orange from the terminal,
+use
+
+	orange-canvas
```
Why is this indent misbehaving????? :angry:

kernc replied:
Tell your editor to replace tabs with spaces.

ajdapretnar commented:
I've fixed and updated things. I wasn't messing with the default Sphinx setup for obvious reasons. Also, I tried to set VERSION to be read from setup.py, but it is a wider problem. Imp module is now deprecated and I can't find a proper function in importlib to substitute imp.load_source function. If you have any ideas, you're more than welcome.

kernc commented:
https://stackoverflow.com/questions/19009932/import-abitrary-python-source-file-python-3-3

ajdapretnar commented:
Should be fixed now. Give it a look.

kernc commented on README.rst:
```diff
@@ -1,12 +1,47 @@
+
+Windows users have to compile the code. Run
+
+    python setup.py build_ext -i --compiler=msvc install
```
I get this error.
```text
running build_ext
running build_src
build_src
building extension "orangecontrib.network._fr_layout" sources
build_src: building npy-pkg config files
No module named 'numpy.distutils.msvccompiler' in numpy.distutils; trying from distutils
customize MSVCCompiler
customize MSVCCompiler using build_ext
No module named 'numpy.distutils.msvccompiler' in numpy.distutils; trying from distutils
customize MSVCCompiler
Missing compiler_cxx fix for MSVCCompiler
customize MSVCCompiler using build_ext
building 'orangecontrib.network._fr_layout' extension
compiling C sources
error: Microsoft Visual C++ 10.0 is required (Unable to find vcvarsall.bat).
```

ajdapretnar replied:
Yes, you still need a compiler to do it. Either Visual Studio or Mingw32, I can add that to the description, if you want, but this is just an eternal pain with Windows where there are no right answers. Even if I write it, there's a good chance it won't work for some people.

kernc replied:
If the compiler is MinGW, the flag is still `--compiler msvc`? If not, I suggest setting it to mingw as that's lighter and easier to install. And mentioning that a compiler is needed seems reasonable. Even on GNU/Linux, the user needs to have a compiler package installed (which is, albeit with `apt-get install g++` much easier, usually not the default case).

ajdapretnar commented:
@kernc Ping! When you have the time, this is ready for another review.

ajdapretnar commented:
@kernc Ping! Ping! Ping! :balloon:

kernc commented on README.rst:
```diff
@@ -1,12 +1,51 @@
+
+or `Mingw32 <https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/rubenvb/gcc-4.8-release/>`__:
+
+    python seutp.py build_ext -i --compiler=mingw32 install
```
typo.

kernc commented on README.rst:
```diff
@@ -1,12 +1,51 @@
+To register this add-on with Orange, but keep the code in the development directory (do not copy it to 
+Python's site-packages directory), run
+
+    python setup.py develop
```
It's better to run
```bash
pip install -e .
```
as that leaves a trail one can uninstall (i.e. `pip uninstall ...` works).

ajdapretnar replied:
Can change. However, this is in **all** current READMEs of Orange. It is in all our documentation web-wide. So I'd prefer to just add `pip install -e .` as an option with explanation to it.

ajdapretnar replied:
Added `pip install -e .` as an option.

kernc commented on doc/conf.py:
```diff
@@ -109,27 +109,30 @@
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+html_theme = 'alabaster'
```
This theme just totally sucks. It's _the worst_. The shitty narrow column likens a blog. Breaks even 50 columns of code in all our examples. Total rubbish.

ajdapretnar replied:
The only way you can say this theme is shitty is to suggest a better alternative.

ajdapretnar replied:
Sticking with alabaster for now as core Orange uses it as well.

kernc commented on doc/widgets/networkanalysis.rst:
```diff
@@ -0,0 +1,20 @@
+-  (None)
+
+Description
+-----------
```
So, umm, why, basically, do we accept nothing here? :smiley:

kernc commented on doc/widgets/networkclustering.rst:
```diff
@@ -0,0 +1,20 @@
+
+**Inputs**:
+
+-  (None)
```
None is not even correct.

kernc commented on doc/widgets/networkexplorer.rst:
```diff
@@ -0,0 +1,20 @@
+
+**Inputs**:
+
+-  (None)
```
Why not TODO then?

ajdapretnar replied:
Please read the PR description. It's all TODO. I explained it already.

kernc commented on docs/rst/code/network-constructor-nx.py:
```diff
@@ -1,29 +0,0 @@
-
-# plot vertices
-plt.plot(x, y, 'ro')
-plt.savefig("network-constructor-nx.png")
```
So these examples went out because they don't work anymore?

ajdapretnar replied:
Because nobody knows whether they work or not. Once the setup for the documentation is done, which is what this PR is all about, I'll start writing proper docs.

ajdapretnar commented:
This is ready for another review. Addressed the changed where deemed appropriate. Again, this is not the full documentation, it's the skeleton I'll be using for future proper docs, so a quick merge would be great. Thanks!
